src/lib/y2storage/partition_id.rb

```diff
@@ -158,5 +158,14 @@ def is?(*names)
         end
       end
     end
+
+    # List of partition ids suitable for a specific partition table.
+    #
+    # @param ptable [Y2Storage::PartitionTables::Base]
+    #
+    # @return [Array<PartitionId>]
+    def self.all_for_partition_table(ptable)
```

Comment:
Wouldn't it be way more natural to have this method/responsibility in the partition table instead of a class method here?

I mean, doesn't `a_device.partition_table.supported_partition_ids` look more natural than `Y2Storage::PartitionId.all_for_partition_table(a_device.partition_table)`?

Reply:
Oh, whatever. To me it looks the same. :-)

Reply:
It's not only more natural, also sorter in general. :-)

Class methods are a resource to be used with care in API design. If it's possible to have an instance method, the resulting API will almost always be more object oriented. 

This is specially true when you have `PartitionTable#partition_id_supported?`. If the instance of that class has the responsibility of knowing if a partition id is supported, it makes sense that such instance also has the responsibility of knowing all the supported ids.